specification/securityinsights/resource-manager/common/1.0/types.json

```diff
@@ -36,7 +36,21 @@
       "x-ms-external": true
     },
     "ResourceWithEtag": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
       "description": "An azure resource object with an Etag property",
+      "properties": {
+        "etag": {
+          "description": "Etag of the azure resource",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
```

Comment:
Can you use a different model name rather than `Resource` which is same as the one in the common-types/types.json? Having duplicated name will cause conflicts when reference the common-types/types.json.

Reply:
This would cause breaking changes in the APIs in the current version which already reference this type.
In any case, we don't reference the common-types/type.json Resource. We have our own as we were requested to add systemData to each resource by ARM but the Resource in the common-types doesn't have it.

Reply:
I saw the reference to common-types for systemData. https://github.com/Azure/azure-rest-api-specs/blob/ce3de83555be8a55abc59ba7d10ba1208d750dd1/specification/securityinsights/resource-manager/common/1.0/types.json#L74

`Resource` model is a new model you added in this PR, rename it won't be a breaking change.

Reply:
Synced with @anat-gilenson offline, the change was already merged in another PR and `Resource` is referenced by other [model](https://github.com/Azure/azure-rest-api-specs/blob/b9b11ff3a46acf33eb207442069e881130215d75/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-04-01/Incidents.json#L1104). @anat-gilenson confirmed this defect will be resolved in next version of types.json.